### PR TITLE
Made Advanced Solar Panel slightly more difficult and added Rubber Synthesizer

### PR DESCRIPTION
**Pull request by NCBPFluffyBear**
Replaced the solar panel in the LX advanced solar panel's recipe with the Slimefun advanced solar generator. 

The advanced solar generator requires 12 solar panels to craft (compared to the original 1 required), 16 more aluminum ingots, 4 more electric motors, and 1 more redstone. Have fun!

### CODE REVIEW

**Comment by WalshyDev** [comment]
Is this easier or harder then the energised one? Whichever one is harder is what I want

**Comment by WalshyDev** [comment]
Also, fix your git user already. It is using BuildTools, you need to set your username and email in relation to your GH ones. See this: https://docs.github.com/en/github/getting-started-with-github/set-up-git#setting-up-git

**Comment by J3fftw1** [comment]
> 
> 
> Is this easier or harder then the energised one? Whichever one is harder is what I want

we could make it even harder and include our glass forgot the name

**Comment by NCBPFluffyBear** [comment]
ok xd ill change it to energized

> we could make it even harder and include our glass forgot the name

Already does

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java [hunk 1]
```diff
@@ -4,10 +4,7 @@
 import dev.j3fftw.litexpansion.items.FoodSynthesizer;
 import dev.j3fftw.litexpansion.items.MagThor;
 import dev.j3fftw.litexpansion.items.Thorium;
-import dev.j3fftw.litexpansion.machine.AdvancedSolarPanel;
-import dev.j3fftw.litexpansion.machine.MassFabricator;
-import dev.j3fftw.litexpansion.machine.RefinedSmeltery;
-import dev.j3fftw.litexpansion.machine.ScrapMachine;
+import dev.j3fftw.litexpansion.machine.*;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: No *

Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: dont wildcard import

**Comment by NCBPFluffyBear** [comment]
https://streamable.com/lmwpo4 
I first showed the bug with the scrap machine (It first takes one in, does nothing, and the process actually starts with the second item) and then showed the synth at work along with testing some possible dupes

File: src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    private void tick(@Nonnull Block b) {
+        @Nullable final BlockMenu inv = BlockStorage.getInventory(b);
+        if (inv == null) return;
+
+        @Nullable final ItemStack input = inv.getItemInSlot(INPUT_SLOT);
+        @Nullable final ItemStack output = inv.getItemInSlot(OUTPUT_SLOT);
+        if (input == null || input.getType() == Material.AIR
+                || !SlimefunUtils.isItemSimilar(input, OIL_BUCKET, true)
+                || (output != null
+                && (output.getType() != Items.RUBBER.getType() || output.getAmount() == output.getMaxStackSize()
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java]: || output.getAmount() == output.getMaxStackSize()
on new line

File: src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import org.bukkit.NamespacedKey;
+import org.bukkit.block.Block;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.Map;
+
+import static io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems.*;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java]: Don't static import and especially not * monkaS

File: src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+public class RubberSynthesizer extends SlimefunItem implements InventoryBlock, EnergyNetComponent {
+
+    public static final RecipeType RECIPE_TYPE = new RecipeType(
+            new NamespacedKey(LiteXpansion.getInstance(), "scrap_machine"), Items.RUBBER_SYNTHESIZER_MACHINE
+    );
+
+    public static final int ENERGY_CONSUMPTION = 20_000 / 26;
```
Comment (WalshyDev) [inline src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java]: This should / PROGRESS_AMOUNT

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Review by WalshyDev — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/ItemSetup.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: Awww he's learning. So proud :^)

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/ItemSetup.java]: i can review aswell a little XD

File: src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java (hunk 3 above)
Comment (NCBPFluffyBear) [reply src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java]: No static no work

Comment (WalshyDev) [reply src/main/java/dev/j3fftw/litexpansion/machine/RubberSynthesizer.java]: Just...... add the class name....

Static imports just turn `Class.CONSTANT` -> `CONSTANT`

**Comment by NCBPFluffyBear** [comment]
I ran the command so the commit messages will be different in the future

**Pull request merged**